test/distributed/fsdp/test_fsdp_misc.py

```diff
@@ -640,6 +640,33 @@ def __init__(self):
         with self.assertRaisesRegex(RuntimeError, "please pass in device_id"):
             FSDP(CPUGPUModule())
 
+    @skip_if_lt_x_gpu(2)
+    def test_fsdp_ignored_module_meta(self):
+        class CPUGPUModule(nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.a = nn.Linear(1, 1)
+                self.b = nn.Linear(1, 1)
+
+        with torch.device("meta"):
+            m = CPUGPUModule()
+        m = FSDP(m, device_id=self.rank, ignored_modules=[m.a], use_orig_params=True)
+        meta_device = torch.device("meta")
+        self.assertEqual(meta_device, next(m.a.parameters()).device)
+
+        # Test with param_init_fn
+        with torch.device("meta"):
+            m = CPUGPUModule()
+        m = FSDP(
+            m,
+            device_id=self.rank,
+            ignored_modules=[m.a],
+            use_orig_params=True,
+            param_init_fn=lambda m: m.cuda(),
```

Comment:
In general, this `lambda m: m.cuda()` would do some repeated checks trying to move modules to CUDA since `param_init_fn` would be called on every module. This should just lead to some CPU overhead since copying to CUDA is a no-op if already on CUDA.

As a tiny nit, the variable shadowing of `m` is also a bit precarious.